test/gtest/uct/ib/test_ib_md.cc

```diff
@@ -362,6 +368,41 @@ UCS_TEST_SKIP_COND_P(test_ib_md, mt_fail,
     }
 }
 
+UCS_TEST_SKIP_COND_P(test_ib_md, mem_window,
+                     !check_invalidate_support(UCT_MD_MEM_ACCESS_RMA))
+{
+    unsigned flags = UCT_MD_FLAG_INVALIDATE_RMA | UCT_MD_FLAG_INVALIDATE_AMO;
+    std::vector<uint8_t> buffer(1024);
+    uct_mem_h base;
+    EXPECT_UCS_OK(reg_mem(UCT_MD_MEM_ACCESS_RMA, buffer.data(), buffer.size(),
+                          &base));
+
+    /* Test case 1: creating MW from memh before mkey_pack */
+    uct_mem_h mw1 = reg_mem_window(base);
+
+    /* Test case 2: creating MW from memh after mkey_pack */
+    std::vector<uint8_t> base_rkey1 = mkey_pack(base, flags);
+    uct_mem_h mw2 = reg_mem_window(base);
+    std::vector<uint8_t> mw2_rkey1 = mkey_pack(mw2, flags);
+    EXPECT_EQ(base_rkey1, mw2_rkey1);
+
+    /* Test case 3: subsequent mkey_pack calls return the same result */
+    std::vector<uint8_t> mw2_rkey2 = mkey_pack(mw2, flags);
+    EXPECT_EQ(mw2_rkey1, mw2_rkey2);
+
+    /* Test case 4: base memh can still be used to pack mkeys */
+    std::vector<uint8_t> base_rkey2 = mkey_pack(base, flags);
+    EXPECT_NE(base_rkey1, base_rkey2);
```

Comment:
i do not think it is correct. base memh should not be modified once derived memh is created

Reply:
ok, fixed this